validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorSourceFactory.java

```diff
@@ -0,0 +1,155 @@
+      addMutableValidatorSource().ifPresent(validatorSources::add);
+    }
+    return validatorSources;
+  }
+
+  public Optional<ValidatorSource> getMutableValidatorSource() {
+    return mutableValidatorSource;
+  }
+
+  private Optional<ValidatorSource> addMutableValidatorSource() {
+    if (maybeDataDir.isEmpty()) {
+      return Optional.empty();
+    }
+    final DataDirLayout dataDirLayout = maybeDataDir.get();
+    final Path keystorePath = ValidatorClientService.getAlterableKeystorePath(dataDirLayout);
+    final Path keystorePasswordPath =
+        ValidatorClientService.getAlterableKeystorePasswordPath(dataDirLayout);
+    if (!(Files.exists(keystorePath) && Files.exists(keystorePasswordPath))) {
+      return Optional.empty();
+    }
+
+    final KeyStoreFilesLocator keyStoreFilesLocator =
+        new KeyStoreFilesLocator(
+            List.of(keystorePath + File.pathSeparator + keystorePasswordPath), File.pathSeparator);
+
+    mutableValidatorSource =
+        Optional.of(
+            mutableSlashingProtected(
+                new LocalValidatorSource(
+                    spec,
+                    config.isValidatorKeystoreLockingEnabled(),
+                    new KeystoreLocker(),
+                    keyStoreFilesLocator,
+                    asyncRunner,
+                    false)));
+    return mutableValidatorSource;
+  }
+
+  private Optional<ValidatorSource> addLocalValidatorSource() {
+    if (config.getValidatorKeys() == null) {
+      return Optional.empty();
+    }
+    KeyStoreFilesLocator keyStoreFilesLocator =
+        new KeyStoreFilesLocator(config.getValidatorKeys(), File.pathSeparator);
+    return Optional.of(
+        slashingProtected(
+            new LocalValidatorSource(
+                spec,
+                config.isValidatorKeystoreLockingEnabled(),
+                new KeystoreLocker(),
+                keyStoreFilesLocator,
+                asyncRunner,
+                true)));
+  }
+
+  private Optional<ValidatorSource> addExternalValidatorSource() {
+    if (config.getValidatorExternalSignerPublicKeySources().isEmpty()) {
+      return Optional.empty();
+    }
+    final ValidatorSource externalValidatorSource =
+        ExternalValidatorSource.create(
+            spec,
+            metricsSystem,
+            config,
+            externalSignerHttpClientFactory,
+            publicKeyLoader,
+            asyncRunner);
+    return Optional.of(
+        config.isValidatorExternalSignerSlashingProtectionEnabled()
+            ? slashingProtected(externalValidatorSource)
+            : externalValidatorSource);
+  }
+
+  private ValidatorSource slashingProtected(final ValidatorSource validatorSource) {
+    return new SlashingProtectedValidatorSource(validatorSource, slashingProtector);
+  }
+
+  private ValidatorSource mutableSlashingProtected(final ValidatorSource validatorSource) {
+    return new MutableLocalValidatorSource(validatorSource, slashingProtector);
+  }
```

Comment:
I think I'd inline this method.  It's not a generic kind of wrapper but specifically for the mutable local validator source.